meilisearch/index.py

```diff
@@ -729,12 +730,15 @@ def delete_document(self, document_id: Union[str, int]) -> TaskInfo:
         )
         return TaskInfo(**response)
 
-    def delete_documents(self, ids: List[Union[str, int]]) -> TaskInfo:
+    def delete_documents(
+        self,
+        ids: List[Union[str, int]],
```

Comment:
Following the new approach from the central issue, the idea should be to move the `delete_documents_by_filter` to the same `delete_documents` by adding a new argument `filter` to the method. Also, I don't know how Python handles deprecations, but the `ids` argument should be deprecated in this case.

Reply:
@brunoocasali I considered this new approach, but think it could be confusing. If a user sends both ids and a filter the filter will be silently ignored. Is this OK? Based on the message this means eventually you will only be able to delete by filter and not id?

Reply:
Yes, that's correct! 

I understand that could be strange at first sight, but the idea is to add the functionality and not break the current usage (without creating a new method just for a new use case).

Also, I understand that our SDKs (most of them) are not stable, but that is not the foresee I have for them. I believe breakings are always a bad sign, especially when they are not "expected" please remember the context explained [here](https://github.com/meilisearch/integration-guides/issues/264).

Since those SDKs are unstable, we can eliminate the mixed behavior when we move them to a v1 (not necessarily being directly related to the engine being on v2). Still, the cool thing is that we introduced warnings/deprecations to the users before doing it, so everyone is "ready".